src/org/loklak/client/SearchClient.java

```diff
@@ -31,39 +31,95 @@
+     */
+    public static Timeline search(final String hostServerUrl,
+                                  final String query,
+                                  final Timeline.Order order,
+                                  final String source,
+                                  final int count,
+                                  final int timezoneOffset,
+                                  final long timeout) throws IOException {
+        Timeline timeline = new Timeline(order);
+        String searchApiUrl = hostServerUrl
+                + SEARCH_API
+                + URLEncoder.encode(query.replace(' ', '+'), ENCODING)
+                + PARAM_TIMEZONE_OFFSET + timezoneOffset
+                + PARAM_MAXIMUM_RECORDS + count
+                + PARAM_SOURCE + (source == null ? PARAM_SOURCE_VALUE : source)
+                + PARAM_MINIFIED + PARAM_MINIFIED_VALUE
+                + PARAM_TIMEOUT + timeout;
+
+        JSONObject json = JsonIO.loadJson(searchApiUrl);
+
+        if (json.length() == 0)
+            return timeline;
+
+        JSONArray statuses = json.getJSONArray(JSON_KEY_STATUSES);
         if (statuses != null) {
             for (int i = 0; i < statuses.length(); i++) {
                 JSONObject tweet = statuses.getJSONObject(i);
-                JSONObject user = tweet.getJSONObject("user");
-                if (user == null) continue;
-                tweet.remove("user");
-                UserEntry u = new UserEntry(user);
-                MessageEntry t = new MessageEntry(tweet);
-                tl.add(t, u);
+                JSONObject user = tweet.getJSONObject(JSON_KEY_USER);
+                if (user == null)
+                    continue;
+                tweet.remove(JSON_KEY_USER);
+                UserEntry userEntry = new UserEntry(user);
+                MessageEntry messageEntry = new MessageEntry(tweet);
+                timeline.add(messageEntry, userEntry);
             }
         }
-        if (json.has("search_metadata")) {
-            JSONObject metadata = json.getJSONObject("search_metadata");
-            if (metadata.has("hits")) {
-                tl.setHits((Integer) metadata.get("hits"));
+
+        if (json.has(JSON_KEY_SEARCH_METADATA)) {
+            JSONObject metadata = json.getJSONObject(JSON_KEY_SEARCH_METADATA);
+            if (metadata.has(JSON_KEY_HITS)) {
+                timeline.setHits((Integer) metadata.get(JSON_KEY_HITS));
             }
-            if (metadata.has("scraperInfo")) {
-                String scraperInfo = (String) metadata.get("scraperInfo");
-                tl.setScraperInfo(scraperInfo);
+            if (metadata.has(JSON_KEY_SCRAPER_INFO)) {
+                String scraperInfo = (String) metadata.get(JSON_KEY_SCRAPER_INFO);
+                timeline.setScraperInfo(scraperInfo);
             }
         }
         //System.out.println(parser.text());
-        return tl;
+        return timeline;
     }
 
```

Comment:
After the end of a method a blank line is left (in the middle of source code also), so following that I left a blank line.